### PR TITLE
Update row in datastore

### PR DESCRIPTION
**Pull request by FenrirUnbound**
Updates an entry in the DynamoDB table by `id`.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -96,8 +96,37 @@ class Dynamodb extends Datastore {
...
+        }
+
+        userData.id = id;
+
+        return client.update(userData, (err, data) => {
+            if (err) {
+                return callback(err);
+            }
+
+            return callback(err, data.toJSON());
```
Comment (nkatzman) [inline index.js:133]: does data contain just the updated fields?

Comment (FenrirUnbound) [inline index.js:133]: Contains everything in the row

**Comment by nkatzman** [comment]
👍

**Comment by tkyi** [comment]
update README plz

**Comment by FenrirUnbound** [comment]
README updated 😸

File: test/index.test.js [hunk 2]
```diff
@@ -256,4 +257,107 @@ describe('index test', () => {
...
+
+            datastore.update({
+                table: 'pipelines',
+                params: {
+                    id,
+                    data: { targetKey: 'updatedValue' }
+                }
+            }, (err, data) => {
+                assert.isNotOk(err);
+                assert.deepEqual(expectedResult, data);
```
Comment (d2lam) [inline test/index.test.js]: Shouldn't it be `actual, expected`
`assert.deepEqual(data, expectedResult)`

File: test/index.test.js [hunk 3]
```diff
@@ -256,4 +257,107 @@ describe('index test', () => {
...
+                }, {
+                    expected: {
+                        id
+                    }
+                });
+                done();
+            });
+        });
+
+        it('fails when given an unknown table name', (done) => {
```
Comment (d2lam) [inline test/index.test.js]: This behavior is not the same as imdb. Which one do we want to go with?

Comment (d2lam) [inline test/index.test.js]: imdb returns callback(null,null) when table doesn't exist 

https://github.com/screwdriver-cd/screwdriver-datastore-imdb/blob/master/index.js#L96-L100

File: index.js [hunk 4]
```diff
@@ -96,8 +96,44 @@ class Dynamodb extends Datastore {
...
+            const err = new Error(`Invalid table name "${config.table}"`);
+
+            return callback(err);
+        }
+
+        userData.id = id;
+
+        return client.update(userData, updateOptions, (err, data) => {
+            if (err) {
+                if (err.statusCode === 400) {
```
Comment (tkyi) [inline index.js:126]: In the future we might want to revisit error handling when the `update` returns 400 from "The conditional request failed" vs. the ID doesn't already exist

File: test/index.test.js [hunk 5]
```diff
@@ -256,4 +257,107 @@ describe('index test', () => {
...
+
+        it('returns nothing when given an unknown table name', (done) => {
+            datastore.update({
+                table: 'doesNotExist',
+                params: {
+                    id: 'doesNotMatter',
+                    data: {}
+                }
+            }, (err, data) => {
+                assert.isNotOk(err);
```
Comment (tkyi) [inline test/index.test.js]: isNull?

**Comment by tkyi** [comment]
👍  okay for now

**Pull request merged**